File: 2017-18/contest1/prog/junior/ninety_nine.java
import java.io.*;
import java.util.*;

public class ninety_nine {
	public static void main(String[] args) throws IOException {
		Scanner input = new Scanner(System.in);
		for(int n = 0; n < 5; ++n) { // read 5 input cases
			int total = input.nextInt();
			int[] cards = new int[10];
			for(int i = 0; i < 10; ++i)
				cards[i] = input.nextInt();
			input.nextLine();

			Queue<Integer> playerHand = new LinkedList<>();
			playerHand.add(cards[0]);
			playerHand.add(cards[1]);
			playerHand.add(cards[2]);

			for(int i = 3; i < 10; ++i) {
				if(i % 2 == 1) { // player's turn
					int card = playerHand.poll();
					switch(card) {
					case 9:
						break; // pass
					case 4:
						total -= 10;
						break;
					case 0:
						if(total + 11 <= 99)
							total += 11;
						else
							total += 1;
						break;
					default:
						total += card;
						break;
					}
					if(total > 99) {
						System.out.println(total + ", dealer");
						break;
					}
					playerHand.add(cards[i]);
				} else { // dealer's turn
					int card = cards[i];
					switch(card) {
					case 9:
						break; // pass
					case 4:
						total -= 10;
						break;
					case 0:
						if(total + 11 <= 99)
							total += 11;
						else
							total += 1;
						break;
					default:
						total += card;
						break;
					}
					if(total > 99) {
						System.out.println(total + ", player");
						break;
					}
				}
			}
		}
	}
}

File: 2017-18/contest1/prog/junior/ninety_nine.py
import os
import sys
from collections import deque

for n in range(5):
	nums = [int(x) for x in input().split()] # read a series of integers from one line
	total = nums[0]
	cards = nums[1:]

	playerHand = deque();
	playerHand.append(cards[0]); # enq on the right
	playerHand.append(cards[1]); # enq on the right
	playerHand.append(cards[2]); # enq on the right

	for i in range(3,10):
		if i % 2 == 1: # player's turn
			card = playerHand.popleft() # deq on the left
			if card == 9:
				pass
			elif card == 4:
				total -= 10
			elif card == 0:
				if total + 11 <= 99:
					total += 11
				else:
					total += 1
			else:
				total += card
			if total > 99:
				print(str(total) + ", dealer")
				break
			playerHand.append(cards[i])
		else: # dealer's turn
			card = cards[i]
			if card == 9:
				pass
			elif card == 4:
				total -= 10
			elif card == 0:
				if total + 11 <= 99:
					total += 11
				else:
					total += 1
			else:
				total += card
			if total > 99:
				print(str(total) + ", player")
				break
